### PR TITLE
docker support

### PR DESCRIPTION
**Pull request by haq**
# Description

Added support for building and publishing a docker image to [GitHub container registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-docker-registry). 
If accepted, you will need to create a secret named `PACKAGES_ACCESS_TOKEN`.
Furthermore, an update will be required to the documentation to include instructions on how to run the image.

```sh
docker volume create lodestone
```

```sh
docker run -d \
  --name lodestone \
  --restart unless-stopped \
  -p 16662:16662 \
  -v lodestone:/root/.lodestone \
  ghcr.io/Lodestone-Team/lodestone_core
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

*Note: make sure your files are formatted with rust-analyzer*

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Generated by Cargo
 # will have compiled files and executables
-/target/
```
Comment (pinnouse) [inline .gitignore:3]: Why do we need the `/` removed?

**Review by pinnouse — COMMENTED** [review]
Any reason for preference of GitHub's Container Registry over say Docker Hub?

**Comment by haq** [comment]
> Any reason for preference of GitHub's Container Registry over say Docker Hub?

Since the source is hosted on GitHub, no reason to not host the images in the same place.

**Review by pinnouse — APPROVED** [review]
LGTM. Will announce when we get the first docker image pushed to registry.

**Pull request merged**